### PR TITLE
Migrate end-to-end tests to GeckoDriver and a modern Firefox version

### PR DESCRIPTION
**Pull request by davehunt**

### CODE REVIEW

File: e2e-tests/README.md [hunk 1]
```diff
@@ -104,10 +110,6 @@ version may not be compatible. Upgrading each of them to latest often fixes it.
 Tips and tricks
 ---------------
 
-Because Selenium opens real browser windows, it will steal focus and switch
-workspaces. Firefox doesn't have a headless mode of operation, so we can't
-simply turn off the UI.
```
Comment (davehunt) [inline e2e-tests/README.md]: GeckoDriver/Marionette does not steal focus by default.

**Comment by willkg** [comment]
We don't want to merge anything until we're done with Elasticsearch migration testing. Marking the PR accordingly so we don't accidentally merge it.

**Comment by willkg** [comment]
@chartjes Is this good to go?

**Comment by chartjes** [comment]
DO IT

**Pull request merged**